Modules/posixmodule.c

```diff
@@ -5381,38 +5379,12 @@ parse_file_actions(PyObject *file_actions,
     return -1;
 }
 
-/*[clinic input]
-
-os.posix_spawn
-    path: path_t
-        Path of executable file.
-    argv: object
-        Tuple or list of strings.
-    env: object
-        Dictionary of strings mapping to strings.
-    /
-    *
-    file_actions: object(c_default='NULL') = ()
-        A sequence of file action tuples.
-    setpgroup: object = NULL
-        The pgroup to use with the POSIX_SPAWN_SETPGROUP flag.
-    resetids: bool(accept={int}) = False
-        If the value is `True` the POSIX_SPAWN_RESETIDS will be activated.
-    setsigmask: object(c_default='NULL') = ()
-        The sigmask to use with the POSIX_SPAWN_SETSIGMASK flag.
-    setsigdef: object(c_default='NULL') = ()
-        The sigmask to use with the POSIX_SPAWN_SETSIGDEF flag.
-    scheduler: object = NULL
-        A tuple with the scheduler policy (optional) and parameters.
-
-Execute the program specified by path in a new process.
-[clinic start generated code]*/
 
 static PyObject *
-os_posix_spawn_impl(PyObject *module, path_t *path, PyObject *argv,
+_posix_spawn(PyObject *module, path_t *path, PyObject *argv,
```

Comment:
I suggest the name "use_posix_spawnp" for the parameter name. I would prefer to add the new parameter as the first parameter. I also propose the rename your function "py_posix_spawn". All in one:

```
static PyObject *
py_posix_spawn(int use_posix_spawnp, ...)
```

"_posix_spawn" sounds like a private name and might conflict with another name, whereas "py_" prefix should only be used by Python.

Reply:
Or "use_path".